src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,39 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+
+        if (!dateIsBeforeOrEqual(dateFrom, dateTo)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        Map<String, Integer> salaries = new LinkedHashMap<>();
```

Comment:
let's do this without Map. Hint: you can append all information inside your loops

Reply:
done)